README.md

```diff
@@ -493,6 +493,38 @@ std::variant<fine::Ok<int64_t>, fine::Error<std::string>> find_meaning(ErlNifEnv
 Note that if you use a particular union frequently, it may be convenient
 to define a type alias with `using`/`typedef` to keep signatures brief.
 
+## Allocators
+
+When using NIFs, memory allocations are sometimes required, but Erlang
+has no knowledge of memory obtained through `new` or `malloc`. For
+Erlang to know about memory consumption in NIFs, `enif_alloc` and
+`enif_free` must be used.
+
+For compatibility with the STL, fine provides both a polymorphic 
+memory resource and a templated allocator:
+
+```c++
+std::vector<std::pmr::string, fine::Allocator<std::pmr::string>> repeat_string(
+    ErlNifEnv *,
+    std::basic_string<char, std::char_traits<char>, fine::Allocator<char>>
+        string,
+    std::uint64_t repeat) {
+  std::vector<std::pmr::string, fine::Allocator<std::pmr::string>> strings;
+
+  for (std::uint64_t i = 0; i != repeat; ++i) {
+    strings.emplace_back(std::pmr::string(string, fine::memory_resource));
+  }
+
+  return strings;
+}
+```
+
+Since `Decoder` has no special overloads for `std::pmr::polymorphic_allocator`,
+this means that attempting to use containers like `std::pmr::string`
+as NIF arguments will call the default constructor of `std::pmr::polymorphic_allocator`,
+which makes use of the `std::pmr::get_default_resource()` memory
+resource.
+
```

Comment:
Should  I add a specialization so that decoding containers like `std::pmr::string` defaults to using `fine::memory_resource` for its `std::pmr::polymorphic_allocator`?